brd-ios/breadwallet/src/Redesign/Scenes/Buy/BuyCoordinator.swift

```diff
@@ -28,19 +33,71 @@ class BuyCoordinator: BaseCoordinator, BuyRoutes, BillingAddressRoutes, OrderPre
         }
     }
     
-    func showThreeDSecure(url: URL) {
+    func showThreeDSecure(url: URL, flow: Flow) {
         let webViewController = SimpleWebViewController(url: url)
         webViewController.setup(with: .init(title: "3D Secure")) // TODO: Localize
         webViewController.didDismiss = { [weak self] in
-            let vc = self?.navigationController.children.first(where: { $0 is BillingAddressViewController }) as? BillingAddressViewController
-            vc?.interactor?.checkThreeDSecureStatus(viewAction: .init())
+            switch flow {
+            case .addCard:
+                let vc = self?.navigationController.children.first(where: { $0 is BillingAddressViewController }) as? BillingAddressViewController
+                vc?.interactor?.checkThreeDSecureStatus(viewAction: .init())
```

Comment:
if ud used the existing fetch cycle, you would not need this magic here (considering the VC's dont need any vars set 2 work)